File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {

        String cleanedString = s.replaceAll("[^a-zA-Z0-9]", "").toLowerCase();
        char[] charArray = cleanedString.toCharArray();
        int left = 0;
        int right = charArray.length - 1;
        while(left < right){
            if(charArray[left] == charArray[right]){
                left += 1;
                right -= 1;
            }
            else{
                return false;
            }
        }
        return true;
    }
}

File: 125-valid-palindrome/valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        

        cleaned = ""
        for char in s:
            if char.isalnum():
                cleaned += char
        cleaned = cleaned.lower()

        left = 0
        right = len(cleaned)-1
        while left<=right:
            if cleaned[left] == cleaned[right]:
                left += 1
                right -= 1
            else:
                return False
        return True